### PR TITLE
14866 updates to ownership for executor scenarios

### PR DESCRIPTION
**Pull request by owillborn**
*Issue #:* /bcgov/entity#14866

*Description of changes:*
- Updates to home owner tenancy type and group tenancy type in executor/trustee/admin scenarios
- Highlighted scenarios here: https://projects.invisionapp.com/share/2C133LCZGDW9#/screens/470980264_Step3_-_UPDATED_TENANCY

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts [hunk 1]
```diff
@@ -112,6 +124,22 @@ export function useHomeOwners (isMhrTransfer: boolean = false) {
     }
   }
 
+  const hasETAGroupError = (group: MhrRegistrationHomeOwnerGroupIF): boolean => {
+    // Variable to track if group has a valid combination of Executor/Trustee/Admin (ETA) Owners
+    let hasETAGroupError = false
+    // If a group contains multiple ETA's within one group, set invalid
+    if (group.owners.filter(owner => executorTrusteeAdmin.includes(owner.partyType)).length > 1) {
+      console.log('Multiple ETA Error')
+      hasETAGroupError = true
```
Comment (dimak1) [inline ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts]: Here we could just `return true` and skip the second condition check below.

File: ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts [hunk 2]
```diff
@@ -112,6 +124,22 @@ export function useHomeOwners (isMhrTransfer: boolean = false) {
     }
   }
 
+  const hasETAGroupError = (group: MhrRegistrationHomeOwnerGroupIF): boolean => {
+    // Variable to track if group has a valid combination of Executor/Trustee/Admin (ETA) Owners
+    let hasETAGroupError = false
```
Comment (dimak1) [inline ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts]: We don't really need this variable, as we can simply return the boolean.

File: ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts [hunk 3]
```diff
@@ -76,6 +82,12 @@ export function useHomeOwners (isMhrTransfer: boolean = false) {
   const getHomeTenancyType = (): HomeTenancyTypes => {
     // Groups
     const groups = getTransferOrRegistrationHomeOwnerGroups().filter(owner => owner.action !== ActionTypes.REMOVED)
+    // Variable to track if owners has a valid combination of Executor/Trustee/Admin (ETA) Owners
+    let hasETAError = false
+    groups.forEach(group => {
+      if (hasETAGroupError(group)) hasETAError = true
```
Comment (dimak1) [inline ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts]: Maybe once we get the ETA error we don't have to continue iterating thru groups?

Comment (owillborn) [reply ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts]: refactored to use arr.some() instead

File: ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts (hunk 2 above)
Comment (owillborn) [reply ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts]: fixed

File: ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts (hunk 1 above)
Comment (owillborn) [reply ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts]: fixed

File: ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts [hunk 4]
```diff
@@ -48,6 +48,12 @@ export function useHomeOwners (isMhrTransfer: boolean = false) {
     'setMhrTransferHomeOwnerGroups'
   ])
 
+  const executorTrusteeAdmin = [
+    HomeOwnerPartyTypes.EXECUTOR,
+    HomeOwnerPartyTypes.TRUSTEE,
+    HomeOwnerPartyTypes.ADMINISTRATOR
+  ]
```
Comment (cameron-eyds) [inline ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts]: Unless using this elsewhere, i would define within the scope of the function expression it's used.

File: ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts [hunk 5]
```diff
@@ -93,7 +103,7 @@ export function useHomeOwners (isMhrTransfer: boolean = false) {
       // Added second condition, because when an owner exists as a Sole Ownership, editing and clicking Done,
       // will change status to Tenants in Common unless above logic is in place..
       return HomeTenancyTypes.SOLE
-    } else if (numOfOwners > 1) {
+    } else if (numOfOwners > 1 && !hasETAError) {
```
Comment (cameron-eyds) [inline ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts]: Just curious how this validation rule ties into determining the Home Tenancy Type. 
By the logic, it implies that if there is ever ever an `ETAError` that we always fall back to a `NA` type? 
This was a requirement?

File: ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts [hunk 6]
```diff
@@ -103,7 +113,7 @@ export function useHomeOwners (isMhrTransfer: boolean = false) {
   const getGroupTenancyType = (group: MhrRegistrationHomeOwnerGroupIF): HomeTenancyTypes => {
     const numOfOwnersInGroup = group.owners.filter(owner => owner.action !== ActionTypes.REMOVED).length
 
-    if (numOfOwnersInGroup > 1) {
+    if (numOfOwnersInGroup > 1 && !hasETAGroupError(group)) {
```
Comment (cameron-eyds) [inline ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts]: Same as above, designs call for Tenancy Type of NA in any scenario where there is a E T or A?

File: ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts [hunk 7]
```diff
@@ -112,6 +122,17 @@ export function useHomeOwners (isMhrTransfer: boolean = false) {
...
+    // If a group contains multiple ETA's within one group, return invalid
+    if (group.owners.filter(owner => executorTrusteeAdmin.includes(owner.partyType)).length > 1) {
+      return true
+    } else
+    // If group has a mix of ETA and living owner, return invalid
+    if (group.owners.some(owner => executorTrusteeAdmin.includes(owner.partyType)) &&
+        group.owners.some(owner => !executorTrusteeAdmin.includes(owner.partyType))) {
+      return true
+    } else return false
+  }
```
Comment (cameron-eyds) [inline ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts]: I believe you can simplify this to ..
```
return (group.owners.filter(owner => executorTrusteeAdmin.includes(owner.partyType)).length > 1) ||
(group.owners.some(owner => executorTrusteeAdmin.includes(owner.partyType)) &&
        group.owners.some(owner => !executorTrusteeAdmin.includes(owner.partyType)))(group.owners.some(owner => executorTrusteeAdmin.includes(owner.partyType)) &&
        group.owners.some(owner => !executorTrusteeAdmin.includes(owner.partyType)))
```
Though doesn't the second condition always resolve as false? They look mutually exclusive no, as in if one is true the other is false and vice versa?

File: ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts (hunk 5 above)
Comment (owillborn) [reply ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts]: The only case where there can be an ETA and not have tenancy type of NA would be Sole Ownership. So I think I could remove the hasETAError and replace with hasETA, checking if an ETA is just present. Thanks for calling this out!

File: ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts (hunk 7 above)
Comment (cameron-eyds) [reply ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts]: What were the findings here?

**Review by cameron-eyds — APPROVED** [review]
Looks good! Thank you for the updates

File: ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts (hunk 7 above)
Comment (owillborn) [reply ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts]: They are mutually exclusive as you suggested. Scrapped that function and replaced in most recent commit

Comment (cameron-eyds) [reply ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts]: oh right on, yes looks good, nice work 👍

**Review by dimak1 — APPROVED** [review]

**Pull request merged**